subprojects/core/src/main/java/org/gradle/api/internal/AbstractTask.java

```diff
@@ -385,7 +385,7 @@ public boolean isSatisfiedBy(Task element) {
     }
 
     @Override
-    public DescribingAndSpec<? super TaskInternal> getOnlyIf() {
+    public Spec<? super TaskInternal> getOnlyIf() {
```

Comment:
Keeping the `DescribingAndSpec`-returning method in the sources will lead to plugins using this API and compiled against Gradle >= 7.6 becoming incompatible with Gradle < 7.6 runtime for no good reason, do you think we can accept this?
Also, yes, there are several community plugins overriding `getOnlyIf` in the tasks.

Maybe we should get better at warning users about internal APIs leaking through `DefaultTask`, so they are aware of the stuff they should not use or override.

Reply:
>will lead to plugins using this API and compiled against Gradle >= 7.6 becoming incompatible with Gradle < 7.6 runtime for no good reason

Oh, I see, I didn't think through this aspect of backwards compatibility. I was mostly trying to stick to the javac-blessed pattern to avoid unforeseen problems.